Feature: Allow block_network.allowed_hosts configuration via fixture

AndreCimander opened the pull request:
### Description

Extended `pytest_recording.plugin.block_network` to allow configuration of allowed_hosts via `vcr_config` fixture.

I had to version pin werkzeug<2.1.0 to avoid https://github.com/postmanlabs/httpbin/issues/673, not sure if you would like to move this change to a separate PR :) 

### Checklist

- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Added a changelog entry
- [x] Extended the README / documentation, if necessary

Stranger6667 commented on docs/changelog.rst:
```diff
@@ -6,6 +6,8 @@ Changelog
 `Unreleased`_
 -------------
 
+- Allow ``block_network.allowed_hosts`` configuration via ``vcr_config`` fixture. #82
```
```suggestion
- Allow ``block_network.allowed_hosts`` configuration via ``vcr_config`` fixture. `#82`_
```

And you need to add an extra entry at the bottom of this file

Stranger6667 commented:
Yep, looks good to me! A small syntactic issue in the changelog

Re: Werkzeug - I am ok with having this change here :)

Stranger6667 commented:
Thanks for bringing this change! :)

AndreCimander commented:
Updated the changelog.

((Pycharm`s local history tried to revert your change suggestion :eyes: ))

Stranger6667 commented:
Seems like another tox job requires pinning werkzeug

AndreCimander commented:
Updated deps for failing tox job, also updated black version in `pre-commit-config.yaml` to 22.3.0 in order to handle click compatibility.

https://github.com/psf/black/issues/2976

Stranger6667 commented:
Thanks! I'll take a look at codecov integration tomorrow

Stranger6667 commented on src/pytest_recording/plugin.py:
```diff
@@ -94,17 +94,25 @@ def block_network(request: SubRequest, record_mode: str, vcr_markers: List[Mark]
     marker = request.node.get_closest_marker(name="block_network")
     if marker is not None:
         validate_block_network_mark(marker)
+    config = request.getfixturevalue("vcr_config")
```
This change makes the `vcr_config` fixture eagerly evaluated for every test. Generally, I tend to avoid evaluating user-defined fixtures until they are needed. Maybe some `allowed_hosts` fixture would work as well - it could be evaluated in the conditional branch below.

Stranger6667 replied:
Though, it could be done separately

Stranger6667 commented:
Thank you @AndreCimander and sorry for the delay!
